posts/polyglot.md

```diff
@@ -0,0 +1,109 @@
+
+If you are working with a team on a project which has a particular culture and approach to their code, you should attempt to work within the style that the rest of the team finds comfortable. Only at this point is it wise to begin to influence changes, and gain support from the team to adopt those changes. 
+
+Learning new cultures is by far the trickiest and requires a high degree of humility.
+
+## My polyglot toolbox
+
+Remembering all the differences between languages is hard work. Being a "specialist" in a language requires serious dedication, internalising knowledge about every corner of a language requires a lot of practical experience.
+
+So how do I remember all of this? The secret is: **I don't**. 
+
+Here, I'll tell you how I get by without being a specialist.
+
+### Be familiar with the paradigms
+
+It's important to learn new languages. The biggest piece of advice to those looking to become a polyglot is to pick languages that are vastly different to what you currently find comfortable.
+
+Working in Ruby? Choose a statically-typed OO language.
+
+Know a statically typed OO language? Learn a functional language.
+
+Know a functional language? Learn a logical programming language.
+
+Know a logical programing language? Learn an old programming language.
+
+Know an old programming language? Learn a new programming language.
+
+Know a new language? Learn a systems programming language?
+And so on.
+
+The point here is comfortability - it is important to recognise that the difference between `C# and Java` (or `Ruby and Python`) is not significant enough to provide paradigm-changing learnings. Pick languages far outside your comfortable norm.
+
+### Find a mentor
+
+The importance of advice from someone who has worked with a language, in a team, has absolutely no comparison. There are tiny details of how "normal code" looks within a particular language that you can only pick up through osmosis and direct feedback from a mentor. 
+
+I have found that meetups e.g. London Code Dojo or London Software Craftsmanship are full of individuals who can provide mentoring over a kata. The subtle transfer of knowledge that takes place in less than 2 hours is substantial enough to be worth your time.
+
+### TDD
+
+The discipline with the biggest impact on my polyglot workflow is TDD. When I say TDD, I mean full on - Disciplined TDD. 
+
+TDD is a strict discipline where a programmer is not able to write a single line of production code without writing a failing test. Given the tests must be written, and they must be written first when TDD'ing a programmer is also only allowed to write the simplest assertion to produce a failing test. When a programmer has a failing test, they are also only allowed to write the simplest production code to make that test pass. What this results in in practice is that a TDD'ing programmer's test suite assertions have to be watertight. 
```

Comment:
"What this yields in practice is a test suite composed of watertight assertions."